pennylane/measurements/var.py

```diff
@@ -51,4 +56,68 @@ def circuit(x):
     """
     if not op.is_hermitian:
         warnings.warn(f"{op.name} might not be hermitian.")
-    return MeasurementProcess(Variance, obs=op)
+    return _Variance(Variance, obs=op)
+
+
+class _Variance(SampleMeasurement, StateMeasurement):
+    """Measurement process that computes the variance of the supplied observable."""
+
+    def process_samples(
+        self,
+        samples: Sequence[complex],
+        wire_order: Wires,
+        shot_range: Tuple[int] = None,
+        bin_size: int = None,
+    ):
+        if isinstance(self.obs, Projector):
+            # branch specifically to handle the projector observable
+            idx = int("".join(str(i) for i in self.obs.parameters[0]), 2)
+            # we use ``self.wires`` instead of ``self.obs`` because the observable was
+            # already applied before the sampling
+            probs = qml.probs(wires=self.wires).process_samples(
+                samples=samples, wire_order=wire_order, shot_range=shot_range, bin_size=bin_size
+            )
+            return probs[idx] - probs[idx] ** 2
+
+        # estimate the variance
+        samples = qml.sample(op=self.obs).process_samples(
+            samples=samples, wire_order=wire_order, shot_range=shot_range, bin_size=bin_size
+        )
+        # With broadcasting, we want to take the variance over axis 1, which is the -1st/-2nd with/
+        # without bin_size. Without broadcasting, axis 0 is the -1st/-2nd with/without bin_size
+        axis = -1 if bin_size is None else -2
+        # TODO: do we need to squeeze here? Maybe remove with new return types
+        return qml.math.squeeze(qml.math.var(samples, axis=axis))
+
+    def process_state(self, state: Sequence[complex], wire_order: Wires):
+        if isinstance(self.obs, Projector):
+            # branch specifically to handle the projector observable
+            idx = int("".join(str(i) for i in self.obs.parameters[0]), 2)
+            # we use ``self.wires`` instead of ``self.obs`` because the observable was
+            # already applied to the state
+            probs = qml.probs(wires=self.wires).process_state(state=state, wire_order=wire_order)
+            return probs[idx] - probs[idx] ** 2
+
+        eigvals = qml.math.asarray(self.obs.eigvals(), dtype=float)
+
+        # the probability vector must be permuted to account for the permuted wire order of the observable
+        old_obs = self.obs
+        new_obs_wires = self._permute_wires(self.obs.wires)
+        self.obs = qml.map_wires(self.obs, dict(zip(self.obs.wires, new_obs_wires)))
+        # we use ``self.wires`` instead of ``self.obs`` because the observable was
+        # already applied to the state
+        prob = qml.probs(wires=self.wires).process_state(state=state, wire_order=wire_order)
+        self.obs = old_obs
```

Comment:
Again, we could just pass `new_obs_wires` to `qml.probs`.

Reply:
Done!